src/state/digital-exchange/categories/actions.js

```diff
@@ -0,0 +1,38 @@
+import { SET_DE_CATEGORIES, SET_SELECTED_DE_CATEGORY } from 'state/digital-exchange/categories/types';
+import { addErrors } from '@entando/messages';
+import { toggleLoading } from 'state/loading/actions';
+
+import { getDECategories } from 'api/digital-exchange/categories';
+
+
+export const setSelectedDECategory = digitalExchangeCategory => ({
+  type: SET_SELECTED_DE_CATEGORY,
+  payload: {
+    digitalExchangeCategory,
+  },
+});
+
+export const setDECategories = digitalExchangeCategories => ({
+  type: SET_DE_CATEGORIES,
+  payload: {
+    digitalExchangeCategories,
+  },
+});
+
+export const fetchDECategories = () => dispatch => (
+  new Promise((resolve) => {
+    dispatch(toggleLoading('digital-exchange/categories'));
+    getDECategories().then((response) => {
+      response.json().then((data) => {
+        if (response.ok) {
+          dispatch(setDECategories(data.payload));
+          dispatch(toggleLoading('digital-exchange/categories'));
```

Comment:
can we move the `dispatch(toggleLoading('digital-exchange/categories'));` outside the if / else ?